### PR TITLE
Generating SDK for Azure Load Testing

### PR DESCRIPTION
**Pull request by NiveditJain**

### CODE REVIEW

File: sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/Program.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+using Azure.Identity;
+using System.IO.Pipes;
+using System.Text.Json.Serialization;
+using Newtonsoft.Json;
+using System.Net;
+using System.Net.Mime;
+
+namespace SampleCodes
+{
+    internal class Program
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/Program.cs:17]: Is there something here that isn't already covered by the samples in `/samples`?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/sample.jmx [hunk 2]
```diff
@@ -0,0 +1,192 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/sample.jmx:1]: did you intend to check this in?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/tests/Samples/Sample1_HelloWorld.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/tests/Samples/Sample1_HelloWorld.cs:1]: let's remove the snippet region until we have a real sample.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_HelloWorld.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# <scenario_list>
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
+
+## <scenario>
+
+You can create a client and call the client's `<operation>` method
+
+```C# Snippet:Azure_Developer_LoadTesting_Scenario
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_HelloWorld.md:9]: remove for now, until we have actual sample code

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+using System.Text;
+using Azure.Core.Pipeline;
+using Azure.Core;
+
+namespace Azure.Developer.LoadTesting
+{
+    /// <summary> The LoadTesting service client. </summary>
+    public partial class LoadTestingClient
+    {
+        private static readonly string[] AuthorizationScopes = new string[] { "https://loadtest.azure-dev.com/.default" };
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs:17]: is `azure-dev.com` the correct public scope?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs:1]: by typical convention, non-generated client types would go in the root of `/src`

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+// Replace with test project/test project public key and uncomment to make internal members visible to
+// your test project. If not needed, this can be deleted.
+// [assembly: InternalsVisibleTo("Azure.Developer.LoadTesting.Generated.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100d15ddcb29688295338af4b7686603fe614abd555e09efba8fb88ee09e1f7b1ccaeed2e8f823fa9eef3fdd60217fc012ea67d2479751a0b8c087a4185541b851bd8b16f8d91b840e51b1cb0ba6fe647997e57429265e85ef62d565db50a69ae1647d54d7bd855e4db3d8a91510e5bcbd0edfbbecaa20a7bd9ae74593daa7b11b4")]
+
+// Replace Microsoft.Test with the correct resource provider namepace for your service and uncomment.
+// See https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/src/properties/AssemblyInfo.cs:11]: remove?

File: sdk/loadtestservice/test-resources.json [hunk 8]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (christothes) [inline sdk/loadtestservice/test-resources.json:1]: replace with real ARM template?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/Program.cs (hunk 1 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/Program.cs]: @christothes earlier I was writing Samples as a different project itself, shifted to /samples folder as suggested in the docs.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/sample.jmx (hunk 2 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/SampleCodes/sample.jmx]: I am not sure if I understood your question correctly, can you please elaborate?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/tests/Samples/Sample1_HelloWorld.cs (hunk 3 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/tests/Samples/Sample1_HelloWorld.cs:1]: Removed samples folder from tests, I hope it is fine?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_HelloWorld.md (hunk 4 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_HelloWorld.md]: Added samples here

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/properties/AssemblyInfo.cs (hunk 7 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/src/properties/AssemblyInfo.cs:11]: done

File: sdk/loadtestservice/test-resources.json (hunk 8 above)
Comment (NiveditJain) [reply sdk/loadtestservice/test-resources.json:1]: @christothes I apologies, I am not able to understand this comment, can you please explain in more detail?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
+# Azure LoadTesting client library for .NET
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:1]: @NiveditJain can you please add a space and make it "Azure Load Testing"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs (hunk 6 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/src/CustomCode/LoadTestingClient.cs]: changed

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+Azure Load Testing provides client library in .net to the user by which they can interact natively with Azure Load Testing service. Azure Load Testing is a fully managed load-testing service that enables you to generate high-scale load. The service simulates traffic for your applications, regardless of where they're hosted. Developers, testers, and quality assurance (QA) engineers can use it to optimize application performance, scalability, or capacity.
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/README.md]: updated!

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md (hunk 9 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/README.md]: updated!

File: sdk/loadtestservice/test-resources.json (hunk 8 above)
Comment (christothes) [reply sdk/loadtestservice/test-resources.json:1]: It appears that this only defines a KeyVault resource. I expected to see an Azure Load Testing resource that would be the target for live tests

**Comment by NiveditJain** [comment]
Dear reviewers ( @annelo-msft @pshao25 @chidozieononiwu @weshaggard @benbp @KrzysztofCwalina @tg-msft @JoshLove-msft @christothes @pallavit @jsquire), 

We have added pending items, awaiting your review. 

Thanks and Regards!

File: sdk/loadtestservice/Azure.Developer.LoadTesting/tests/LoadTestingClientTest.cs [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+namespace Azure.Developer.LoadTesting.Tests
+{
+    public class LoadTestingClientTest: RecordedTestBase<LoadTestingClientTestEnvironment>
+    {
+        public string testId;
+        public string fileId;
+        public string testRunId;
+        public string appComponentId;
+
+        public LoadTestingClientTest(bool isAsync) : base(isAsync, RecordedTestMode.Record)
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/tests/LoadTestingClientTest.cs:26]: Based on the build errors, it looks like you need a body sanitizer for the storage sas tokens

here is an example: 
https://github.com/Azure/azure-sdk-for-net/blob/bd5c715e0b25f80fee7e78f779c1cab9f6387f50/sdk/storage/Azure.Storage.Blobs.Batch/tests/BlobBatchClientTests.cs#L34-L36

File: sdk/loadtestservice/Azure.Developer.LoadTesting/tests/LoadTestingClientTestEnvironment.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.TestFramework;
+
+namespace Azure.Developer.LoadTesting.Tests
+{
+    public class LoadTestingClientTestEnvironment : TestEnvironment
+    {
+        public string Endpoint => GetRecordedVariable("ENDPOINT");
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/tests/LoadTestingClientTestEnvironment.cs:11]: Best practice is to use a service specific name 
```suggestion
        public string Endpoint => GetRecordedVariable("LOADTESTSERVICE_ENDPOINT");
```

Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/tests/LoadTestingClientTestEnvironment.cs]: Resolved

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/Azure.Developer.LoadTesting.csproj [hunk 13]
```diff
@@ -16,7 +16,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
-    <PackageReference Include="System.Text.Json" />
+    <PackageReference Include="System.Text.Json" VersionOverride="6.0.6" />
```
Comment (weshaggard) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/src/Azure.Developer.LoadTesting.csproj:19]: Why do we need to override this version? In general we should only do this in exception cases.

Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/src/Azure.Developer.LoadTesting.csproj:19]: @weshaggard, actually we wanted to use System.Text.Json.Nodes which is not available in the default version, in our preview release we won't be using it, so we removed `VersionOverride` for now.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/LoadTestAdministrationClient.cs [hunk 14]
```diff
@@ -0,0 +1,344 @@
...
+
+                using HttpMessage message = CreateUploadTestFileRequest(testId, fileId, RequestContent.Create(body), fileType, context);
+                message.Request.Headers.SetValue("Content-Type", "multipart/form-data;boundary=" + boundary);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
```
Comment (christothes) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/src/LoadTestAdministrationClient.cs:283]: Can we use [MultiPartContent](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/Multipart/MultipartContent.cs) here? [Here](https://github.com/Azure/azure-sdk-for-net/blob/e29cd16283b0213e99b41376cf08542d03d6640a/sdk/tables/Azure.Data.Tables/src/TableRestClient.cs#L55) is an example of its use

File: sdk/loadtestservice/Azure.Developer.LoadTesting/CHANGELOG.md [hunk 15]
```diff
@@ -0,0 +1,11 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
```
Comment (chunyu3) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/CHANGELOG.md:3]: Please write the changelog. please refer to changelog guideline: https://azure.github.io/azure-sdk/policies_releases.html#change-logs

Write the changes and fit to right blocks(Features Added, Breaking Changes and so on). If the change block does not contain content, remove the block.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample01_CreateOrUpdateTest.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
+# Create or Update Load Test
```
Comment (chunyu3) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample01_CreateOrUpdateTest.md:1]: please follow our guide to write the samples https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/Create_Samples_Guidance.md. Thanks

And this is our sample RP https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template/Azure.Template FYI

File: sdk/loadtestservice/Azure.Developer.LoadTesting/CHANGELOG.md (hunk 15 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/CHANGELOG.md:3]: corrected changelog!

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample01_CreateOrUpdateTest.md (hunk 16 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample01_CreateOrUpdateTest.md]: Resolved

**Review by chunyu3 — APPROVED** [review]
Just one suggestion: update the Snippet name to follow the format: {NamespaceInUnderscore}_{Scenario}, such as Azure_Developer_LoadTesting_CreateOrUpdateTest

**Review by ninallam — APPROVED** [review]

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+  [Source code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/src) | [Package (NuGet)](https://www.nuget.org/packages?q=Azure.Developer.Loadtesting) | [API reference documentation](https://azure.github.io/azure-sdk-for-net) | [Product documentation](https://learn.microsoft.com/azure/load-testing/)
+
+
+## Documentation
+
+Various documentation is available to help you get started
+
+- [API reference documentation][api_reference_doc]
+- [Product Documentation][product_documentation]
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:13]: @NiveditJain these links may not work. Can we please update?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+### Authenticate the client
+
+To use an [Azure Active Directory (AAD) token credential][authenticate_with_token],
+provide an instance of the desired credential type obtained from the
+[azure-identity][azure_identity_credentials] library.
+
+To authenticate with AAD, you must first use [nuget][nuget] install [`azure-identity`][azure_identity_nuget]
+
+After setup, you can choose which type of [credential][azure_identity_credentials] from Azure.Identity to use.
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:38]: Can you check these links as well?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+## Key concepts
+
+The following components make up the Azure Load Testing service. The Azure Load Test client library for C# allows you to interact with each of these components through the use of clients. There is a top level client  `LoadTestingClient` with two sub-clients
+
+- `LoadTestAdministrationClient`
+
+- `TestRunClient`
+
+These sub-clients are used for managing and using different components of the service. For each method in both of these sub-clients there is a corresponding Async method in the same class, with Async with same implimentation howeven enabling async functionalities, example, if there is a method, `CreateOrUpdateTest` as a part of `LoadTestAdministrationClient` then there always exists one more function `CreateOrUpdateTestAsync` in the same client class. 
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:52]: For each method in both of these sub-clients there is a corresponding Async method in the same class, with the same implementation however enabling async functionalities. For example, if there is a method, `CreateOrUpdateTest` as a part of `LoadTestAdministrationClient` then there always exists one more function `CreateOrUpdateTestAsync` in the same client class.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+### Load Test Administration Client
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:54]: Make this a level 4 heading  #### Load Test Administration Client

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+#### Metrics
+
+During a load test, Azure Load Testing collects metrics about the test execution. There are two types of metrics:
+
+1. Client-side metrics give you details reported by the test engine. These metrics include the number of virtual users, the request response time, the number of failed requests, or the number of requests per second.
+
+2. Server-side metrics are available for Azure-hosted applications and provide information about your Azure application components. Metrics can be for the number of database reads, the type of HTTP responses, or container resource consumption.
+
+### Test Run Client
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:74]: Bring this after #### Load Test Administration Client and make it a level 4 heading as well.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+[Handling failures](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#reporting-errors-requestfailedexception) |
+[Diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md) |
+[Mocking](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#mocking) |
+[Client lifetime](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/)
+<!-- CLIENT COMMON BAR -->
+
+## Next steps
+
+Get started with our [samples](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/loadtestservice/Azure.Developer.LoadTesting/samples):
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:117]: Remove the ":" and add "."

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+## Create LoadTestAdministrationClient
+```C# Snippet:Azure_Developer_LoadTesting_CreatingClient
+string endpoint = TestEnvironment.Endpoint;
+TokenCredential credential = TestEnvironment.Credential;
+
+// creating LoadTesting Client
+LoadTestingClient loadTestingClient = new LoadTestingClient(endpoint, credential);
+
+// getting appropirate Subclient
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md:15]: Correct the typo "appropriate"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md [hunk 24]
```diff
@@ -0,0 +1,56 @@
+# Create or Update Load Test 
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
+
+You can create a LoadTestclient and call the `CreateOrUpdateTest` method from SubClient `LoadTestAdministrationClient`
+
+## Create LoadTestAdministrationClient
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md:7]: Change this to "Create LoadTesting"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        engineInstances = 1,
+        splitAllCSVs = false,
+    },
+    secrets = new { },
+    enviornmentVariables = new { },
+    passFailCriteria = new
+    {
+        passFailMetrics = new { },
+    }
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md:39]: Please provide some sample values for secrets, environment variables and pass fail criteria.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTestAsync.md [hunk 26]
```diff
@@ -0,0 +1,56 @@
+# Create or Update Load Test Async
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTestAsync.md:4]: Same feedback as Sample1_CreateOrUpdateTest.md

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFile.md [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+
+## Create LoadTestAdministrationClient
+```C# Snippet:Azure_Developer_LoadTesting_CreatingClient
+string endpoint = TestEnvironment.Endpoint;
+TokenCredential credential = TestEnvironment.Credential;
+
+// creating LoadTesting Client
+LoadTestingClient loadTestingClient = new LoadTestingClient(endpoint, credential);
+
+// getting appropirate Subclient
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFile.md:15]: Correct the typo.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFile.md [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+string fileId = "my-file-id";
+
+try
+{
+    // uploading file
+    Response response = loadTestAdministrationClient.UploadTestFile(testId, fileId, File.OpenRead(
+        Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "sample.jmx")
+        ));
+
+    // if the test is created successfully, printing response
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFile.md:34]: If the file is uploaded successfully

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFileAsync.md [hunk 29]
```diff
@@ -0,0 +1,43 @@
+# Upload JMX File
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample2_UploadTestFileAsync.md:3]: Same feedback as Sample2_UploadTestFile.md

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 30]
```diff
@@ -0,0 +1,58 @@
+# Create or Update App Component
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
+
+You can create a LoadTestclient and call the `CreateOrUpdateAppComponent` method from SubClient `LoadTestAdministrationClient`
+
+## Create LoadTestAdministrationClient
+```C# Snippet:Azure_Developer_LoadTesting_CreatingClient
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:8]: Same feedback as the other files.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 31]
```diff
@@ -0,0 +1,58 @@
...
+
+## Calling CreateOrUpdateAppComponent
+```C# Snippet:Azure_Developer_LoadTesting_CraeteOrUpdateAppComponent
+// provide unique identifier for your test
+string testId = "my-test-id";
+
+// provide unique app component id
+string appComponentId = "my-app-component-id";
+
+string appComponentConnectionString = "/subscriptions/7c71b563-0dc0-4bc0-bcf6-06f8f0516c7a/resourceGroups/App-Service-Sample-Demo-rg/providers/Microsoft.Web/sites/App-Service-Sample-Demo";
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:27]: Use a subscription ID as "00000000-0000-0000-0000-000000000000"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+// all other data to be sent to AppCompoent
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:28]: Correct typo "AppComponent"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 33]
```diff
@@ -0,0 +1,58 @@
...
+    testid = testId,
+    name = "New App Component",
+    value = new
+    {
+        appComponentConnectionString = new
+        {
+            resourceId = appComponentConnectionString,
+            resourceName = "App-Service-Sample-Demo",
+            resourceType = "Microsoft.Web/sites",
+            subscriptionId = "7c71b563-0dc0-4bc0-bcf6-06f8f0516c7a"
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:40]: Change subscription id to "00000000-0000-0000-0000-000000000000".

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+};
+
+try
+{
+    // uploading file
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:47]: Change the comment

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+    Response response = loadTestAdministrationClient.CreateOrUpdateAppComponents(appComponentId, RequestContent.Create(data));
+    // if the test is created successfully, printing response
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponent.md:49]: Change the comment

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponentAsync.md [hunk 36]
```diff
@@ -0,0 +1,58 @@
+# Create or Update App Component
+
+To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Developer.LoadTesting/README.md#getting-started) for details.
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample3_CreateOrUpdateAppComponentAsync.md:3]: Same feedback

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample4_CreateAndUpdateTest.md [hunk 37]
```diff
@@ -0,0 +1,47 @@
+# Create And Update Test
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample4_CreateAndUpdateTest.md:2]: How is this different from Create or Update test?

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md [hunk 38]
```diff
@@ -0,0 +1,135 @@
...
+The data-plane endpoint is obtained from Control Plane APIs.
+
+**Example:** `1234abcd-12ab-12ab-12ab-123456abcdef.eus.cnt-prod.loadtesting.azure.com`
+
+In the above example, `eus` represents the Azure region `East US`.
+## Examples
+
+You can familiarize yourself with different APIs using [Samples](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/loadtestservice/Azure.Developer.LoadTesting/samples).
+
```
Comment (ninallam) [inline sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:93]: Any reason why we aren't providing samples here directly? Is there a best practice. Want to make sure this is replicated for other languages as well.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md (hunk 17 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:13]: @ninallam these links point to, 
[api_reference_doc]: https://docs.microsoft.com/rest/api/loadtesting/
[product_documentation]: https://azure.microsoft.com/services/load-testing/

Please let me know if you have some other link(s).

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md (hunk 18 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:38]: [azure_identity_nuget]: https://www.nuget.org/packages/Azure.Identity/1.7.0
[nuget]: https://www.nuget.org/
[azure_identity_credentials]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity#credentials

Similarly these links!

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md (hunk 24 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample1_CreateOrUpdateTest.md:7]: @ninallam idea here to show the sample of using correct sub-client, do we want to change it to "LoadTesting" from "LoadTestAdministrationClient"

File: sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample4_CreateAndUpdateTest.md (hunk 37 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/samples/Sample4_CreateAndUpdateTest.md:2]: This is for test run added naming to doc

File: sdk/loadtestservice/Azure.Developer.LoadTesting/README.md (hunk 38 above)
Comment (NiveditJain) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/README.md:93]: I am not sure if it is a best practice, following the standards for C# samples, they are generated from code, so directly adding samples to main Readme could be challenging, however, what is possible is providing a list of links to various sample for quick reference.

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/LoadTestAdministrationClient.cs (hunk 14 above)
Comment (lmazuel) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/src/LoadTestAdministrationClient.cs]: Talked to @christothes , resolving

File: sdk/loadtestservice/Azure.Developer.LoadTesting/src/Azure.Developer.LoadTesting.csproj (hunk 13 above)
Comment (lmazuel) [reply sdk/loadtestservice/Azure.Developer.LoadTesting/src/Azure.Developer.LoadTesting.csproj]: Talked to @christothes that confirmed I can resolve this conversation

**Pull request merged**